src/components/package-port/CodeAndPreview.tsx

```diff
@@ -346,7 +355,10 @@ export function CodeAndPreview({ pkg }: Props) {
     })
   }
 
-  const { handleCreateFile } = useFileManagement(state, setState)
+  const { handleCreateFile, handleDeleteFile } = useFileManagement(
+    state,
+    setState,
```

Comment:
big anti-pattern. `state` is a really bad var name, and it's super unclear from this API how `useFileManagement` is supposed to work.

```suggestion
```

Here is an example of an API that is more appropriate:

```tsx
const { handleCreateFile, handleDeleteFile, handleSave,  fsMap, isLoadingFiles } = usePackageFileManagement({
  package_id: "..."
})
```